src/tests/ftest/launch.py

```diff
@@ -1304,21 +1308,20 @@ def _get_available_interfaces(servers, clients):
             command = " ".join(["cat", os.path.join(net_path, interface, "speed")])
             result = run_remote(logger, all_hosts, command)
             # Verify each host has the same interface speed
-            if result.passed and result.homogeneous:
+            if not result.homogeneous:
+                logger.error(
+                    "Non-homogeneous interface speed detected for %s on %s.",
+                    interface, str(all_hosts))
+            elif not result.passed and "Invalid argument" in "\n".join(result.output[0].stdout):
```

Comment:
I don't think this is going to be consistent across O/Ses.  On some it's `EINVAL`, on others it returns `-1`.  Who knows what it does on yet others.  I think you need to check the driver type of the device to be sure it's a virtio device.  I.e.:

```
# readlink /sys/class/net/eth0/device/driver/module
../../../../module/virtio_net
```

If the base name is `virtio_net`, then you can simply return whatever large value you want as the speed.  Something like:

```
if [[ $(readlink /sys/class/net/eth0/device/driver/module) = */virtio_net ]]; then echo 1000; else cat /sys/class/net/eth0/speed; fi
```

instead of your `cat /sys/class/net/eth0/speed` and then just remove this error handling.

Reply:
I see.  What's the advantage to separating out the virtio check into a separate remote command vs. just doing all of the logic remotely in a single remote command?  It's not a significant difference, indeed.  What I could say is that the latter would be slightly faster and one less round-trip to break.